src/qt/sendcoinsdialog.cpp

```diff
@@ -600,10 +600,11 @@ SendCoinsEntry *SendCoinsDialog::addEntry()
     entry->clear();
     entry->setFocus();
     ui->scrollAreaWidgetContents->resize(ui->scrollAreaWidgetContents->sizeHint());
-    qApp->processEvents();
-    QScrollBar* bar = ui->scrollArea->verticalScrollBar();
-    if(bar)
-        bar->setSliderPosition(bar->maximum());
+    QMetaObject::invokeMethod(ui->scrollArea, [this] {
+        if (ui->scrollArea->verticalScrollBar()) {
+            ui->scrollArea->verticalScrollBar()->setValue(ui->scrollArea->verticalScrollBar()->maximum());
+        }
+    }, Qt::QueuedConnection);
```

Comment:
This is most likely because we are creating the widget in a separate thread.

Try this:
```diff
diff --git a/src/qt/sendcoinsdialog.cpp b/src/qt/sendcoinsdialog.cpp
--- a/src/qt/sendcoinsdialog.cpp	(revision 114d738830368dc155f27a72128adf91b4fffd37)
+++ b/src/qt/sendcoinsdialog.cpp	(date 1672923728012)
@@ -600,11 +600,9 @@
     entry->clear();
     entry->setFocus();
     ui->scrollAreaWidgetContents->resize(ui->scrollAreaWidgetContents->sizeHint());
-    QMetaObject::invokeMethod(ui->scrollArea, [this] {
-        if (ui->scrollArea->verticalScrollBar()) {
-            ui->scrollArea->verticalScrollBar()->setValue(ui->scrollArea->verticalScrollBar()->maximum());
-        }
-    }, Qt::QueuedConnection);
+    if (ui->scrollArea->verticalScrollBar()) {
+        ui->scrollArea->verticalScrollBar()->setValue(ui->scrollArea->verticalScrollBar()->maximum());
+    }
 
     updateTabsAndLabels();
     return entry;
@@ -697,6 +695,13 @@
     return true;
 }
 
+void SendCoinsDialog::showEvent(QShowEvent* event)
+{
+    if (ui->scrollArea->verticalScrollBar()) {
+        ui->scrollArea->verticalScrollBar()->setValue(ui->scrollArea->verticalScrollBar()->maximum());
+    }
+}
+
 void SendCoinsDialog::setBalance(const interfaces::WalletBalances& balances)
 {
     if(model && model->getOptionsModel())
Index: src/qt/sendcoinsdialog.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/qt/sendcoinsdialog.h b/src/qt/sendcoinsdialog.h
--- a/src/qt/sendcoinsdialog.h	(revision 114d738830368dc155f27a72128adf91b4fffd37)
+++ b/src/qt/sendcoinsdialog.h	(date 1672923728023)
@@ -49,6 +49,8 @@
     void pasteEntry(const SendCoinsRecipient &rv);
     bool handlePaymentRequest(const SendCoinsRecipient &recipient);
 
+    void showEvent(QShowEvent* event) override;
+
 public Q_SLOTS:
     void clear();
     void reject() override;
```

Reply:
Thank you for the suggestion!

While this (also) fixes the first segfault, it doesn't achieve the intended scrolling behavior (at least on my macOS setup). 

Before clicking 'Add Recipient':

<img width="799" alt="image" src="https://user-images.githubusercontent.com/116917595/210798920-844108e5-125a-46d8-96c9-752ddf3bcc18.png">

After:

<img width="795" alt="image" src="https://user-images.githubusercontent.com/116917595/210798464-ff4478d9-f9e7-4108-ae94-1f50ae385b00.png">

Whereas the original workaround (and my workaround of the workaround) gives this "After":

<img width="796" alt="image" src="https://user-images.githubusercontent.com/116917595/210797010-681fd76c-f2b5-410c-ac7c-33d205d1b27d.png">

It seems to be a fairly [common](https://stackoverflow.com/questions/65087088/how-to-move-to-end-of-scrollbar-after-adding-qwidget-into-the-qscrollarea) issue [without](https://stackoverflow.com/questions/63884331/scroll-to-the-bottom-when-adding-an-element-to-a-qscrollarea) an [ideal](https://stackoverflow.com/questions/19836404/qscrollarea-does-not-scroll-to-maximum-after-widgets-have-been-added) solution. There are more involved solutions, like subscribing to `rangeChanged` signals, but those would require much more logic (eg - ignore those when manually resizing the viewport, etc.).

Reply:
hmm true, ok. Sounds good to go with the simplest fix here.
Would be good to add a comment above so we don't forget the rationale behind it.

Reply:
Good call. Adding.